HangmanGame/src/main/java/br/edu/iff/bancodepalavras/dominio/tema/emmemoria/MemoriaTemaRepository.java

```diff
@@ -0,0 +1,71 @@
+package main.java.br.edu.iff.bancodepalavras.dominio.tema.emmemoria;
+
+import java.util.HashMap;
+import java.util.Map;
+import main.java.br.edu.iff.bancodepalavras.dominio.tema.Tema;
+import main.java.br.edu.iff.bancodepalavras.dominio.tema.TemaRepository;
+import main.java.br.edu.iff.repository.RepositoryException;
+
+public class MemoriaTemaRepository implements TemaRepository {
+    private static MemoriaTemaRepository soleInstance; // Instância única
+    private final Map<Long, Tema> temaPool; // Pool de temas, usando ID como chave
+
+    // Construtor privado
+    private MemoriaTemaRepository() {
+        this.temaPool = new HashMap<>();
+    }
+
+    // Método Singleton para obter a única instância de MemoriaTemaRepository
+    public static MemoriaTemaRepository getSoleInstance() {
+        if (soleInstance == null) {
+            soleInstance = new MemoriaTemaRepository();
+        }
+        return soleInstance;
+    }
+
+    @Override
+    public synchronized void inserir(Tema tema) throws RepositoryException {
```

Comment:
Precisa também de um verificador para o nome dentro do Tema. Se, por exemplo  o nome for o mesmo, a inserção não deve ser autorizada.
```suggestion
    public synchronized void inserir(Tema tema) throws RepositoryException {
    if (temaPool.containsKey(tema.getId())) ||  (temaPool.containsValue(tema.getNome())) {
    throw new RepositoryException("Tema já existe na pool.");
}

    
```